glean-core/ios/Glean/Glean.swift

```diff
@@ -452,6 +467,21 @@ public class Glean {
         GleanDebugUtility.handleCustomUrl(url: url)
     }
 
+    /// Returns true if running in the base application process, otherwise returns false
+    private func checkIsMainProcess() -> Bool {
+        if isMainProcess == nil {
+            if let packageType = Bundle.main.object(forInfoDictionaryKey: "CFBundlePackageType") as? String {
+                isMainProcess = packageType != "XPC!"
```

Comment:
Can we document/link to what `XPC!` means?
No one will know that this is what extensions are.
Also the way it stands if there are other package types that are not XPC but also not the main app it would report wwrong. Should this instead be an equal test (`==`) to what the package type of the main app is?

Reply:
The only package types we are interested in now fall under XPC services (as app extensions).  [The other possibilites are "APPL" which is an app, "FMWK" for frameworks, and "BNDL", for which "XPC!" is a more specific type of "BNDL".](https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundlepackagetype). [See also](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html#//apple_ref/doc/uid/20001431-111321).  

I'll add some documentation in the form of a comment and I'm happy to invert the `if` statement if it makes you feel more comfortable I just didn't want to exclude potential valid package types so I followed the method of detecting running in an extension with [this code](https://searchfox.org/mozilla-mobile/rev/d5408842b4214f583bd50cdd679addb5464ce8bf/focus-ios/Shared/AppInfo.swift#14-29) taken from Focus.

Reply:
In that case I guess I'm fine with `!= XPC`